frontend/packages/dev-console/src/components/health-checks/create-health-check-probe-utils.ts

```diff
@@ -0,0 +1,91 @@
+import * as _ from 'lodash';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import { HealthCheckProbeData, RequestType, HealthChecksProbeType } from './health-checks-types';
+import { Resources, HealthChecksData } from '../import/import-types';
+import { healthChecksDefaultValues } from './health-check-probe-utils';
+
+export const constructProbeData = (data: HealthCheckProbeData, resourceType?: Resources) => {
+  const probeData = {
+    ...(data.failureThreshold && { failureThreshold: data.failureThreshold }),
+    ...(data.successThreshold && { successThreshold: data.successThreshold }),
+    ...(data.requestType === RequestType.ContainerCommand && {
+      exec: data.exec,
+    }),
+    ...(data.requestType === RequestType.HTTPGET && {
+      httpGet: {
+        ...data[data.requestType],
+        ...(data[data.requestType]?.scheme?.[0] === 'HTTPS' && {
+          scheme: data[data.requestType].scheme[0],
+        }),
+        port: resourceType === Resources.KnativeService ? 0 : _.toInteger(data.httpGet.port),
+      },
+    }),
+    ...(data.requestType === RequestType.TCPSocket && {
+      tcpSocket: {
+        port: resourceType === Resources.KnativeService ? 0 : _.toInteger(data.tcpSocket.port),
+      },
+    }),
+    ...(data.initialDelaySeconds && {
+      initialDelaySeconds: data.initialDelaySeconds,
+    }),
+    ...(data.periodSeconds && { periodSeconds: data.periodSeconds }),
+    ...(data.timeoutSeconds && { timeoutSeconds: data.timeoutSeconds }),
+  };
+  return probeData;
+};
+
+export const getRequestType = (data: HealthCheckProbeData) => {
+  if (_.has(data, RequestType.HTTPGET)) return RequestType.HTTPGET;
+  if (_.has(data, RequestType.TCPSocket)) return RequestType.TCPSocket;
+  if (_.has(data, 'exec.command')) return RequestType.ContainerCommand;
+  return '';
+};
+
+export const getHealthChecksData = (resource: K8sResourceKind): HealthChecksData => {
+  const containers = _.get(resource, 'spec.template.spec.containers', []);
+  const readinessProbe = _.get(containers[0], HealthChecksProbeType.ReadinessProbe, {});
+  const livenessProbe = _.get(containers[0], HealthChecksProbeType.LivenessProbe, {});
+  const startupProbe = _.get(containers[0], HealthChecksProbeType.StartupProbe, {});
```

Comment:
You dont need to use lodash get here. For example-
```suggestion
  const startupProbe =  containers?.[0]?.[HealthChecksProbeType.StartupProbe] ?? {} ;
```